File: examples/thymeleaf-java-example/src/main/java/de/tschuehly/spring/viewcomponent/thymeleaf/application/core/ExampleService.java
package de.tschuehly.spring.viewcomponent.thymeleaf.application.core;

import org.springframework.stereotype.Service;

import java.util.HashMap;
import java.util.NoSuchElementException;

@Service
public class ExampleService {
    Integer dataIndex = 0;
    public HashMap<Integer, String> itemList = new HashMap<>();

    public String getHelloWorld() {
        return "Hello World";
    }

    public void addItemToList(String item) {
        itemList.put(dataIndex, item);
        dataIndex += 1;
    }

    public void deleteItem(Integer id) {
        if (itemList.get(id) == null) {
            throw new NoSuchElementException("No Element with Id: " + id + " found");
        }
        itemList.remove(id);
    }

}

File: examples/thymeleaf-example/src/main/kotlin/de/tschuehly/spring/viewcomponent/thymeleaf/application/core/ExampleService.kt
package de.tschuehly.spring.viewcomponent.thymeleaf.application.core

import org.springframework.stereotype.Service
import java.util.NoSuchElementException

@Service
class ExampleService {
    var dataIndex = 0
    val someData = mutableMapOf<Int, String>()

    fun getHelloWorld(): String {
        return "Hello World"
    }

    fun addItemToList(item: String){
        someData[dataIndex] = item
        dataIndex += 1
    }

    fun deleteItem(id: Int) {
        someData.remove(id) ?: throw NoSuchElementException("No Element with Id: $id found")
    }


}